wandb/sdk/wandb_run.py

```diff
@@ -3013,7 +3013,9 @@ def _log_artifact(
                     is_user_created=is_user_created,
                     use_after_commit=use_after_commit,
                 )
-                artifact._set_save_future(future, self._public_api().client)
+                handle.add_probe(self._on_probe_exit)
+                handle.add_progress(self._on_progress_exit)
```

Comment:
I read the code, but I still have no idea what `_on_probe_exit` and `_on_progress_exit` do. Could you add a docstring?

Reply:
done